### PR TITLE
[WIP] feature: bbn-wallet-connect native lib

### PR DESCRIPTION
**Pull request by gbarkhatov**
The purpose of this PR is to have a working connection with the switch to a native wallet connector via `@babylonlabs-io/bbn-wallet-connect`

This PR is a combination of:
1. https://github.com/babylonlabs-io/simple-staking/pull/404
2. https://github.com/babylonlabs-io/simple-staking/pull/396
3. current `main` changes
4. some temp fixes to make the staking process work

Issues could be solved across different repos:

- https://github.com/babylonlabs-io/simple-staking
- https://github.com/babylonlabs-io/bbn-wallet-connect
- https://github.com/babylonlabs-io/bbn-core-ui

Prioritized:

- Ordinals <- Can be a part of different PR
- Build
  - From main
  - Install lib
  - investigate the context
- Wallets disconnection process

- Connection can be a part of later PRs
- Scroll locker is minor
- Tomo wallet connector is big, should be a part of later PRs
- Connection persistance we're not touching right now

Problems, in terms of priority:
1. Ordinals / Inscriptions will throw an error / not working
2. Connection process is wrong:

- When BTC is selected, it is automatically connected. Should happen after "done" click
- Same story with BBN wallet (Keplr in our case)
- If I select 1 or 2 wallets and click "cancel" the app state is not updated https://i.imgur.com/d2z8XBS.png https://i.imgur.com/uk1UGum.png

3. No TOMO wallet connector integrated
5. When disconnected, the connect button is not clickable https://i.imgur.com/MyA15C7.png
6. Build process keeps failing, both on local and Github CI https://i.imgur.com/Ch22gUU.png
7. `Scroll Locker` is not working. All the connection modals allow user to scroll. This is not the case for a regular app modals. For example, Preview modal is locking the scroll https://i.imgur.com/fV7ugQr.png
8. Connection is not persisted (on refresh need to reconnect once again) <- minor, can be extracted to an issue in https://github.com/babylonlabs-io/bbn-wallet-connect

Overall the **connection works**, I was able to stake using OKX + Keplr

### CODE REVIEW

**Comment by totraev** [comment]
> Connection process is wrong:
When BTC is selected, it is automatically connected. Should happen after "done" click
Same story with BBN wallet (Keplr in our case)
If I select 1 or 2 wallets and click "cancel" the app state is not updated https://i.imgur.com/d2z8XBS.png https://i.imgur.com/uk1UGum.png

@gbarkhatov Implemented according current requirements: 
https://www.figma.com/design/05UW9XBXoFWcKEEIB1ZetV/Design-File?node-id=303-21335&node-type=canvas&m=dev

**Comment by totraev** [comment]
2. Implemented according current requirements, pls check here: https://www.figma.com/design/05UW9XBXoFWcKEEIB1ZetV/Design-File?node-id=303-21335&node-type=canvas&m=dev
6. Don't worry about this for now. I need make `core-ui` a peer dependency of `bbn-wallet-connect` to fix it

File: package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
   "dependencies": {
     "@babylonlabs-io/babylon-proto-ts": "0.0.3-canary.3",
     "@babylonlabs-io/bbn-core-ui": "^0.2.0",
+    "@babylonlabs-io/bbn-wallet-connect": "^0.0.18",
```
[the remark this answers is not in the document]
Comment (gbarkhatov) [reply package.json:30]: @jrwbabylonlab Tomo wallet connect is not currently implemented, it requires additional abstraction

@totraev Should Tomo wallet connect be a dependency and the part of the `simple-staking` or `bbn-wallet-connect`?

**Comment by gbarkhatov** [comment]
> Implemented according current requirements, pls check here:
@totraev according to the design the connection flow assumption in the notes are correct - Once clicked BTC should be "connected" with the address and all of that only within the modal, not within the app

**Comment by totraev** [comment]
@gbarkhatov can be closed in favor of new PR https://github.com/babylonlabs-io/simple-staking/pull/413

**Pull request closed without merging**